includes/admin/class-senseilms-plugin-updater.php

```diff
@@ -0,0 +1,232 @@
+			// Early return in case request to SenseiLMS.com failed.
+			return $res;
+		}
+
+		$res                = new stdClass();
+		$res->name          = $remote->name;
+		$res->slug          = $remote->slug;
+		$res->author        = $remote->author;
+		$res->version       = $remote->version;
+		$res->requires      = $remote->requires;
+		$res->tested        = $remote->tested;
+		$res->requires_php  = $remote->requires_php;
+		$res->last_updated  = $remote->last_updated;
+		$res->sections      = [
+			'description'  => $remote->sections->description,
+			'installation' => $remote->sections->installation,
+			'changelog'    => $remote->sections->changelog,
+		];
+		$res->download_link = $remote->download_url;
+		$res->banners       = [
+			'low'  => $remote->banners->{'1x'},
+			'high' => $remote->banners->{'2x'},
+		];
+
+		// TODO: ADD CSS.
+		wp_enqueue_style( 'sensei-updater-styles', plugin_dir_url( $this->plugin_full_name ) . 'assets/dist/senseilms-licensing/styles.css', [], $this->version );
+
+		return $res;
+	}
+
+	/**
+	 * Potentially injects the details for a new plugin version by checking against the remote server.
+	 * This is done by hooking into the `update_plugins` transient by using the `site_transient_update_plugins` hook.
+	 *
+	 * @param mixed $transient The plugin_update transient.
+	 *
+	 * @hooked site_transient_update_plugins See reference for `site_transient_transient`.
+	 *
+	 * @return mixed
+	 */
+	public function maybe_inject_custom_update_to_update_plugins_transient( $transient ) {
+
+		// Skip empty transients or if it was already set by Sensei Pro.
+		if ( empty( $transient ) || isset( $response[ $this->plugin_full_name ] ) ) {
+			return $transient;
+		}
+
+		$remote = $this->request_info();
+		if ( is_wp_error( $remote ) ) {
+			// Request failed so do not inject anything into the transient.
+			return $transient;
+		}
+
+		if (
+			$remote
+			&& version_compare( $this->version, $remote->version, '<' )
+			&& version_compare( get_bloginfo( 'version' ), $remote->requires, '>=' )
+			&& version_compare( PHP_VERSION, $remote->requires_php, '>=' )
+		) {
+
+			$res                                 = new stdClass();
+			$res->slug                           = $remote->slug;
+			$res->plugin                         = $this->plugin_full_name;
+			$res->new_version                    = $remote->version;
+			$res->tested                         = $remote->tested;
+			$res->package                        = $remote->download_url;
+			$res->icons                          = (array) $remote->icons;
+			$transient->response[ $res->plugin ] = $res;
+		}
+		return $transient;
+	}
+
+	/**
+	 * Helper function that retrieves the latest version information from the remote server if there is a valid license in the system.
+	 * This function caches remote response by using transients.
+	 *
+	 * @return array|WP_Error
+	 */
+	private function request_info() {
+		$cache_key = self::CACHE_KEY_PREFIX . $this->plugin_slug;
```

Comment:
It keeps the same transient we have in Sensei Pro, so if it was already fetched, we don't do it again.